src/parse_fn.rs

```diff
@@ -224,3 +145,103 @@ pub(crate) fn parse_fn_params(tokens: TokenStream) -> Punctuated<FunctionParamet
 
     fields
 }
+
+pub(crate) fn consume_fn_return(tokens: &mut TokenIter) -> Option<([Punct; 2], TyExpr)> {
+    let dash = match tokens.peek() {
+        Some(TokenTree::Punct(punct)) if punct.as_char() == '-' => punct.clone(),
+        _ => return None,
+    };
+    tokens.next().unwrap();
+
+    let tip = match tokens.next() {
+        Some(TokenTree::Punct(punct)) if punct.as_char() == '>' => punct,
+        _ => panic!("cannot parse fn return: expected '>' after '-' token"),
+    };
+
+    Some((
+        [dash, tip],
+        TyExpr {
+            tokens: (consume_stuff_until(
+                tokens,
+                |token| match token {
+                    TokenTree::Group(group) if group.delimiter() == Delimiter::Brace => true,
+                    TokenTree::Ident(i) if i == &Ident::new("where", i.span()) => true,
+                    TokenTree::Punct(punct) if punct.as_char() == ';' => true,
+                    _ => false,
+                },
+                true,
+            )),
+        },
+    ))
+}
+
+/// Tries to parse a function definition.
+///
+/// Panics when the following tokens do not constitute a function definition, with one exception:
+/// when `const` is followed by an identifier which is not `fn`, then `None` is returned. This is to
+/// allow fallback to a constant declaration (both can begin with the `const` token).
+pub(crate) fn try_consume_fn(
```

Comment:
This function should be called `consume_fn`.

Overall, the naming convention so far has been that `consume_xxx` functions in venial return None if the syntax item is absent, and panic if the *beginning* of the syntax item is present but the expect continuation isn't.

Eg: `consume_where_clause` will return None if it doesn't find a `where` token, but will panic if it gets `where "hello world"`.

Reply:
Done.